web3-functions/addRewardsForEthCcReferrals/index.ts

```diff
@@ -0,0 +1,187 @@
+        type: 'bytes32',
+      },
+    ],
+    name: 'ReferralRegistered',
+    type: 'event',
+  },
+] as const
+
+const IDEMPOTENT_REWARD_POOL_ABI = [
+  {
+    inputs: [
+      {
+        components: [
+          {
+            internalType: 'address',
+            name: 'user',
+            type: 'address',
+          },
+          {
+            internalType: 'uint256',
+            name: 'amount',
+            type: 'uint256',
+          },
+          {
+            internalType: 'bytes32',
+            name: 'idempotencyKey',
+            type: 'bytes32',
+          },
+        ],
+        internalType: 'struct IdempotentRewardPool.RewardData[]',
+        name: 'rewards',
+        type: 'tuple[]',
+      },
+      {
+        internalType: 'uint256[]',
+        name: 'rewardFunctionArgs',
+        type: 'uint256[]',
+      },
+    ],
+    name: 'addRewards',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+] as const
+
+Web3Function.onRun(async (context: Web3FunctionEventContext) => {
+  // Get event log from Web3FunctionEventContext
+  const { log, secrets, multiChainProvider } = context
+
+  const gelatoRelayerApiKey = await secrets.get('GELATO_RELAYER_API_KEY')
+  if (!gelatoRelayerApiKey) {
+    return {
+      canExec: false,
+      message: `GELATO_RELAYER_API_KEY not set in secrets`,
+    }
+  }
+  const rewardPoolOwnerMnemonic = await secrets.get(
+    'REWARD_POOL_OWNER_MNEMONIC',
+  )
+  if (!rewardPoolOwnerMnemonic) {
+    return {
+      canExec: false,
+      message: `REWARD_POOL_OWNER_MNEMONIC not set in secrets`,
+    }
+  }
+
+  // Parse event using viem's decodeEventLog
+  const event = decodeEventLog({
+    abi: DIVVI_REGISTRY_ABI,
+    eventName: 'ReferralRegistered',
+    topics: log.topics as [Hex, ...Hex[]],
+    data: log.data as Hex,
+  })
+
+  // Handle event data
+  const { user, rewardsProvider, rewardsConsumer, chainId, txHash } = event.args
+  console.log(
+    `ReferralRegistered: user=${user}, provider=${rewardsProvider}, consumer=${rewardsConsumer}, chainId=${chainId}, txHash=${txHash}`,
+  )
+
+  const relay = new GelatoRelay()
+
+  const account = mnemonicToAccount(rewardPoolOwnerMnemonic as Hex)
+  const client = createWalletClient({
+    account,
+    transport: http(multiChainProvider.default().connection.url),
+  })
+  console.log(account.address)
+
+  if (!isAddressEqual(rewardsProvider, DIVVI_ETHCC2025_REWARDS_ENTITY)) {
+    return {
+      canExec: false,
+      message: `Rewards provider ${rewardsProvider} is not the Divvy EthCC 2025 rewards entity`,
+    }
+  }
+
+  // TODO limit to first X referrals
+
+  const data = encodeFunctionData({
+    abi: IDEMPOTENT_REWARD_POOL_ABI,
+    functionName: 'addRewards',
+    args: [
+      [
+        {
+          user: rewardsConsumer,
+          amount: 1n,
+          // The referrer should only get the reward once, so we use the rewardsConsumer as the idempotency key
+          idempotencyKey: rewardsConsumer,
```

Comment:
...should this be padded out to 32-bytes?  since i'd expect rewardsConsumer to be a 20-byte address.